### PR TITLE
Add fwd declaration headers

### PR DESCRIPTION
**Pull request by strega-nil**
As opposed to declaring these types at each point of use, add header files for this instead. What are your thoughts @cngzhnp ?

### CODE REVIEW

**Comment by cngzhnp** [comment]
Interesting way to do! :) Actually, if there is no rule about how to include/order header files and forward declarations on the project, please define one. (_clang-format_ does something but might not be good enough) If it is already defined and this implementation does not break anything at all, seems good to me! 

On the other hand, I want to say a few words about it. I'd prefer to keep them on header files because I like [include-what-you-use](https://github.com/include-what-you-use/include-what-you-use) principle. 

If you do not want to see repeated forward declarations, there are a few problems with the design. Because when you use them, you try to improve modularity. However, the same forward declarations are done on most of the header files, it might be a design smell. My suggestion is for the next steps, please try to break apart Vcpkgpaths class otherwise these kind of common use will make you have headache or try to implement Pimpl idiom if possible to get rid of things more easily.

TL;DR LGTM, however please consider what I wrote above. Thanks for the request btw!

**Comment by cngzhnp** [comment]
@strega-nil Should I merge them into my branch or will you proceed from another pull request?

**Comment by strega-nil** [comment]
@cngzhnp I proceeded from another pull request.
